### PR TITLE
#547 donation campaign - view my history contributes

### PR DESCRIPTION
**Pull request by vietdangquoc**
# Ticket
#547 

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: frontend/modules/wallet/components/wallet-details/wallet-detail-tabs/shared/transaction-history-card/transaction-history-card.tsx [hunk 1]
```diff
@@ -21,6 +21,27 @@ import { AddressFilterDropdown } from '@/components/ui/address-filter-dropdown';
...
+
+const URL_PARAM = {
+  FROM_FILTER: 'from_filter',
+  TO_FILTER: 'to_filter',
+  TYPE: 'type',
+} as const;
+
+type TransactionTypeValue = (typeof TRANSACTION_TYPE)[keyof typeof TRANSACTION_TYPE];
+type FilterType = (typeof FILTER_TYPE)[keyof typeof FILTER_TYPE];
+
```
Comment (phongvu2812) [inline frontend/modules/wallet/components/wallet-details/wallet-detail-tabs/shared/transaction-history-card/transaction-history-card.tsx]: please put all these constants in the constants.ts

**Review by hoangdohuy — APPROVED** [review]
LGTM
